core/src/main/scala/com/banno/kafka/consumer/ConsumerOps.scala

```diff
@@ -345,13 +347,32 @@ case class ConsumerOps[F[_], K, V](consumer: ConsumerApi[F, K, V]) {
     * failed records when the consumer is closed, and will likely result in less
     * reprocessing after a failure. The consumer must be configured to disable
     * offset auto-commits.
+    *
+    * If keepAliveInterval is provided and finite, and topics have been manually
+    * assigned, then committed offsets will be periodically refreshed to prevent
```

Comment:
This should work with both `assign` and `subscribe` so I think we can remove this constraint from the docs:

> and topics have been manually assigned

Reply:
I'm pretty sure it _isn't_ safe to do this for subscribed rather than assigned topics, at least not without quite a bit more work ... if the topic is subscribed to rather than assigned I believe that it can migrate partitions, which means that the accumulated offset state would become invalid. To track subscribed topics across partitions we would also need to listen for notifications via `ConsumerRebalanceListener`.

Reply:
I also think that it's unnecessary in the subscribed case because the expiry rules are,
> For subscribed consumers, committed offset of a specific partition will be expired and discarded when 1) this retention period has elapsed after the consumer group loses all its consumers (i.e. becomes empty); 2) this retention period has elapsed since the last time an offset is committed for the partition and the group is no longer subscribed to the corresponding topic.

ie. when there are no more consumers, or no more consumers subscribed to the topic in question. In that scenario the periodic refresh would not be running anyway.

Reply:
Excellent points about rebalancing, and being unnecessary when using subscribe. I withdraw my previous comment 😄